CSharp/ArticleProjects/DotNetTutorial/Program.cs

```diff
@@ -288,15 +288,34 @@ private static string GetContainerSasUrl(CloudBlobClient blobClient, string cont
         /// <returns>A <see cref="System.Threading.Tasks.Task"/> object that represents the asynchronous operation.</returns>
         private static async Task CreatePoolAsync(BatchClient batchClient, string poolId, IList<ResourceFile> resourceFiles)
         {
-            Console.WriteLine("Creating pool [{0}]...", poolId);
-
-            // Create the unbound pool. Until we call CloudPool.Commit() or CommitAsync(), no pool is actually created in the
-            // Batch service. This CloudPool instance is therefore considered "unbound," and we can modify its properties.
-            CloudPool pool = batchClient.PoolOperations.CreatePool(
-                poolId: poolId,
-                targetDedicated: 3,                                                         // 3 compute nodes
-                virtualMachineSize: "small",                                                // single-core, 1.75 GB memory, 225 GB disk
-                cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));   // Windows Server 2012 R2
+            CloudPool pool = null;

```

Comment:
Can you take a look at `azure-batch-samples\CSharp\Common\GettingStartedCommon.cs` `CreatePoolIfNotExistAsync` and use similar logic to there to do this if-not-exist check? The `HResult` check is pretty opaque and hard to understand, it's better to have a clear semantic check. I am specifically talking about this:

```
            // Attempt to create the pool
            try
            {
                // Create an in-memory representation of the Batch pool which we would like to create.  We are free to modify/update 
                // this pool object in memory until we commit it to the service via the CommitAsync method.
                Console.WriteLine("Attempting to create pool: {0}", pool.Id);

                // Create the pool on the Batch Service
                await pool.CommitAsync().ConfigureAwait(continueOnCapturedContext: false);

                successfullyCreatedPool = true;
                Console.WriteLine("Created pool {0} with {1} {2} nodes",
                    poolId,
                    poolTargetNodeCount,
                    poolNodeVirtualMachineSize);
            }
            catch (BatchException e)
            {
                // Swallow the specific error code PoolExists since that is expected if the pool already exists
                if (e.RequestInformation != null &&
                    e.RequestInformation.BatchError != null &&
                    e.RequestInformation.BatchError.Code == BatchErrorCodeStrings.PoolExists)
                {
                    // The pool already existed when we tried to create it
                    successfullyCreatedPool = false;
                    Console.WriteLine("The pool already existed when we tried to create it");
                }
                else
                {
                    throw; // Any other exception is unexpected
                }
            }
```

**note**: While it's tempting to just add a project reference and just use the attached method, we want the article project to stand on its own which means it can't have any helper dependencies, so copying this is probably the best way to go.

Also there is a `// If the pool already existed, make sure that its targets are correct` section in that method I mentioned which I think you can skip in this case (it's not as perfectly correct then, but I think for the article project simple is better than 100% correct)